Mathlib/Analysis/Analytic/Constructions.lean

```diff
@@ -1174,7 +1174,17 @@ theorem Finset.analyticWithinAt_prod {A : Type*} [NormedCommRing A] [NormedAlgeb
 
 /-- Finite products of analytic functions are analytic -/
 @[fun_prop]
-theorem Finset.analyticAt_prod {A : Type*} [NormedCommRing A] [NormedAlgebra 𝕜 A]
+theorem Finset.analyticAt_prod {α : Type*} {A : Type*} [NormedCommRing A] [NormedAlgebra 𝕜 A]
+    {f : α → E → A} {c : E} (N : Finset α) (h : ∀ n ∈ N, AnalyticAt 𝕜 (f n) c) :
+    AnalyticAt 𝕜 (∏ n ∈ N, f n) c := by
+  simp_rw [← analyticWithinAt_univ] at h ⊢
+  have : (∏ n ∈ N, f n) = (fun z ↦ ∏ n ∈ N, f n z) := by aesop
+  rw [this]
+  exact N.analyticWithinAt_prod h
+
+/-- Finite products of analytic functions are analytic -/
+@[fun_prop]
+theorem Finset.analyticAt_prod_fun {A : Type*} [NormedCommRing A] [NormedAlgebra 𝕜 A]
```

Comment:
```suggestion
theorem Finset.analyticAt_fun_prod {A : Type*} [NormedCommRing A] [NormedAlgebra 𝕜 A]
```
This should be `fun_prod` because, in the statement, the word `fun` comes before the product.

For coherence of the API, when you fix a lemma like that, it is important to also fix the neighboring ones. Do you think you could rename all the lemmas in this section adding `fun_`, and derive the non-fun version from the fun version by the same `convert` trick as above?

Reply:
Implemented.